#Finding the standard deviation

In [11]:
import pandas as pd
import numpy as np

speed = [60, 70, 80, 90, 100]
x = np.std(speed)
print(x)

14.142135623730951


In [12]:
print(np.mean(speed))

80.0


#Finding the variance

In [13]:
a = np.var(speed)
print(a)


200.0


#Finding the percentile

In [14]:
df = pd.read_csv('pokemon_data.csv')
df = pd.DataFrame(df)
df

#                   Name   Type 1  Type 2  HP  Attack  Defense  \
0      1              Bulbasaur    Grass  Poison  45      49       49   
1      2                Ivysaur    Grass  Poison  60      62       63   
2      3               Venusaur    Grass  Poison  80      82       83   
3      3  VenusaurMega Venusaur    Grass  Poison  80     100      123   
4      4             Charmander     Fire     NaN  39      52       43   
..   ...                    ...      ...     ...  ..     ...      ...   
795  719                Diancie     Rock   Fairy  50     100      150   
796  719    DiancieMega Diancie     Rock   Fairy  50     160      110   
797  720    HoopaHoopa Confined  Psychic   Ghost  80     110       60   
798  720     HoopaHoopa Unbound  Psychic    Dark  80     160       60   
799  721              Volcanion     Fire   Water  80     110      120   

     Sp. Atk  Sp. Def  Speed  Generation  Legendary  
0         65       65     45           1      False  
1         80       80     60           1      False  
2        100      100     80           1      False  
3        122      120     80           1      False  
4         60       50     65           1      False  
..       ...      ...    ...         ...        ...  
795      100      150     50           6       True  
796      160      110    110           6       True  
797      150      130     70           6       True  
798      170      130     80           6       True  
799      130       90     70           6       True  

[800 rows x 12 columns]

In [15]:
perc = np.percentile(df['Defense'], 75)
perc

90.0

In [18]:
perc1 = np.percentile(df['Generation'], 30)
perc1

2.0

#Scaling


I will compare 'Sp.Atk' and 'Generation' columns using z-scaling method' formula. In the dataset it's 170 and 6, which is hard to compare, so I will scale it to same measurement unit before comparing.

In [21]:
mean1 = np.mean(df['Sp. Atk'])
std1 = np.std(df['Sp. Atk'])
z1 = (df['Sp. Atk'].loc[798] - mean1) / std1
z1

2.9716985612059386

In [22]:
mean2 = np.mean(df['Generation'])
std2 = np.std(df['Generation'])
z2 = (df['Generation'].loc[798] - mean2) / std2
z2

1.6119543128755616

Now, when I have these two numbers, I can easily compare them

Here I am scaling all the values in these two columns

In [24]:
from sklearn import linear_model
from sklearn.preprocessing import StandardScaler

scale = StandardScaler()

x = df[['Sp. Atk', 'Generation']]
scaled_x = scale.fit_transform(x)
scaled_x

array([[-0.2391303 , -1.39963712],
       [ 0.21955954, -1.39963712],
       [ 0.83114599, -1.39963712],
       ...,
       [ 2.36011211,  1.61195431],
       [ 2.97169856,  1.61195431],
       [ 1.74852566,  1.61195431]])